src/test/java/com/dawn/jat/illuminati/post/controller/PostControllerTest.java

```diff
@@ -62,34 +58,24 @@ void findPostAll_whenNoRecord() {
 
     @Test
     void findPostAll_whenRecord() {
-        Mockito.when(postService.findAll()).thenReturn(Arrays.asList(postEntity1));
-        assertThat(postController.findAll().size(), is(1));
+        Mockito.when(postService.findAll()).thenReturn(Arrays.asList(postEntity1, postEntity2));
+        assertThat(postController.findAll().size(), is(2));
         Mockito.verify(postService, Mockito.times(1)).findAll();
     }
 
-    @Test
-    public void twoDiffObj_notEqualsSymmetric() {
-        assertFalse(postEntity1.hashCode() == postEntity2.hashCode());
-    }
-
-    @Test
-    public void postEntity_doNotHave_unexpectedValue() {
-        assertFalse(postEntity1.toString().contains("@$*&$abzn&*"));
-    }
-
     @Test
     public void whenPostIdIsAvail_thenRetrievedPostIsCorrect() {
-        Mockito.when(postService.findById("1")).thenReturn(Optional.of(new PostEntity()));
-        ResponseEntity<String> responseEntity = postController.getPostById("1");
+        Mockito.when(postService.findById("1")).thenReturn(Optional.of(postEntity1));
+        ResponseEntity<Object> responseEntity = postController.getPostById("1");
         assertNotNull(responseEntity);
         assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
-        assertNotEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+        assertNotEquals(HttpStatus.NOT_FOUND, responseEntity.getStatusCode());
     }
 
     @Test
-    public void whenPostIdIsUnavail_thenRetrievedPostIsIncorrect() {
-        ResponseEntity<String> responseEntity = postController.getPostById("1A467");
-        assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
-        assertNotEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    void whenPostIdIsUnavail_thenRetrievedPostIsIncorrect() {
+        assertThrows(PostNotfoundException.class, () -> {
+            postController.getPostById("1A467");
```

Comment:
nothing to make sure id is not existed 
try to make mock of controller
```
    PostController mockPostController = mock(PostController.class);
    Mockito.when(mockPostController.getPostById("1A467")).thenThrow(new PostNotfoundException(){});
        assertThrows(PostNotfoundException.class, () -> {
            mockPostController.getPostById("1A467");
        });
```

Reply:
Updated